src/usr/share/netcheck/netcheck.py

```diff
@@ -127,296 +133,295 @@ def _dns_works(self, network):
+        Returns True on successful DNS lookup, False otherwise.
         """
 
-        self.logger.trace('_connect_and_check_dns: Attempting to connect to and reach the '
-                          'Internet over %s.' % network_name)
+        self.logger.trace('_activate_connection_and_check_dns: Attempting to activate and '
+                          'reach the Internet over connection %s.', connection_id)
 
         overall_success = False
-        connection_successful = self.network_meta.connect(network_name)
-        if not(connection_successful):
-            self.logger.debug('_connect_and_check_dns: Could not connect to network %s.' %
-                              network_name)
+        activation_successful = self.network_helper.activate_connection(connection_id)
+        if not activation_successful:
+            self.logger.debug('_activate_connection_and_check_dns: Could not activate '
+                              'connection %s.', connection_id)
         else:
-            self.logger.trace('_connect_and_check_dns: Connected to network %s.' %
-                              network_name)
-            dns_successful = self._dns_works(network_name)
-            if not(dns_successful):
-                self.logger.debug('_connect_and_check_dns: DNS on network %s failed.' %
-                                  network_name)
+            self.logger.trace('_activate_connection_and_check_dns: Connection %s activated.',
+                              connection_id)
+            dns_successful = self._dns_works(connection_id)
+            if not dns_successful:
+                self.logger.debug('_activate_connection_and_check_dns: DNS on connection %s '
+                                  'failed.', connection_id)
             else:
-                self.logger.trace('_connect_and_check_dns: DNS on network %s successful.' %
-                                  network_name)
+                self.logger.trace('_activate_connection_and_check_dns: DNS on connection %s '
+                                  'successful.', connection_id)
                 overall_success = True
 
         return overall_success
 
-    def _check_connection_and_check_dns(self, network_name):
-        """Checks the connection to a network and checks DNS availability if connection
-        exists.
+    def _check_connection_and_check_dns(self, connection_id):
+        """Checks if a connection is active and if successful, checks DNS availability.
 
-        Returns true on success, false on failure.
+        connection_id: The name of the connection as displayed in NetworkManager.
+        Returns True on successful DNS lookup, False otherwise.
         """
         self.logger.trace('_check_connection_and_check_dns: Attempting to reach the '
-                          'Internet over %s.' % network_name)
+                          'Internet over connection %s.', connection_id)
 
         overall_success = False
```

Comment:
We might want to put a try/except block here so we don't get stick on a single connection.